indico/modules/events/forms.py

```diff
@@ -68,3 +68,15 @@ def __init__(self, *args, **kwargs):
 
     def is_submitted(self):
         return super(EmailEventPersonsForm, self).is_submitted() and 'submitted' in request.form
+
+
+class GeneralSettingsReferenceTypeForm(IndicoForm):
+    references = MultipleItemsField(_('External IDs'), [DataRequired()],
+                                    fields=[{'id': 'type', 'caption': _("External IDs"), 'type': 'select'},
+                                            {'id': 'id', 'caption': _("ID"), 'type': 'text'}],
+                                    description=_("Manage external resources"))
+
+    def __init__(self, *args, **kwargs):
+        super(GeneralSettingsReferenceTypeForm, self).__init__(*args, **kwargs)
+        reference_types = ReferenceType.find_all()
+        self.references.fields[0]['choices'] = {e.id: e.name for e in reference_types}
```

Comment:
:volcano: This looks a lot like you are modifying data on the Form class itself! I would rather add `self.choices = {}` to the constructor (so it's a new dict for each field instance in a form) and then set it like this:

    self.references.choices['type'] = {...}